chia/types/blockchain_format/program.py

```diff
@@ -48,6 +48,16 @@ def from_bytes(cls, blob: bytes) -> Program:
     def fromhex(cls, hexstr: str) -> Program:
         return cls.from_bytes(hexstr_to_bytes(hexstr))
 
+    @classmethod
+    def from_json_dict(cls: Type[Program], json_dict: Any) -> Program:
+        if isinstance(json_dict, cls):
+            return json_dict
```

Comment:
this check is odd and unfortunate. but we have at least one test that relies on this behavior, and it mimics what we have on line 187-188 in `streamable.py` prior to this patch